### ISSUE TITLE
Call Hierarchy always returns no results for function aliased via `import as`

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#3893]
## Code Snippet

```python
# declare.py

def func():
    return 1
```

```python
# consume.py

from declare import func
from declare import func as foobar

def callByName():
    func()

def callByAlias():
    foobar()
```

## Repro Steps

1. Create files as above.
2. Right-click, "Show Call Hierarchy" on `func()` call in `consume.py`. See that expected results are displayed.
![image](https://user-images.githubusercontent.com/12820898/216138311-ea93cf7f-0fb7-4480-91b5-06eae9097d5c.png)
4. Do same on `foobar()` call.

## Expected behavior

Call hierarchy shows call to `foobar` in `callByAlias`

## Actual behavior

![image](https://user-images.githubusercontent.com/12820898/216137939-eb320588-22ae-4275-a396-70a9ba902c4c.png)

Initial issue is that the declaration is an Alias and we only support call hierarchies for Function and Class declaration types.

**Comment by debonte** [comment]
We don't close Pylance issues until the fix ships.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023341-29-March-2023-PreRelease)
